core/translate/optimizer.rs

```diff
@@ -119,14 +119,18 @@ fn eliminate_orderby_like_groupby(plan: &mut SelectPlan) -> Result<()> {
     // This function optimizes query execution by eliminating duplicate expressions between ORDER BY and GROUP BY clauses
     // When the same column appears in both clauses, we can avoid redundant sorting operations
     // The function reorders GROUP BY expressions and removes redundant ORDER BY expressions to ensure consistent ordering
-    while order_index < order_by_clauses.len() {
-        let (order_expr, direction) = &order_by_clauses[order_index];
 
-        // Skip descending orders as they require separate sorting
-        if matches!(direction, Direction::Descending) {
-            order_index += 1;
-            continue;
-        }
```

Comment:
actually is this whole function wrong? when can we ever just remove _some_ terms of the order by?

Reply:
oh now i remember, this function has some logic to reorder the group by terms to match the order by... i gotta see if it still makes sense though